libs/vkd3d/utils.c

```diff
@@ -647,6 +648,34 @@ VkFormat vkd3d_get_vk_format(DXGI_FORMAT format)
     return VK_FORMAT_UNDEFINED;
 }
 
+/* get some size-based low bits for prioritization in the same way as d3d12 */
+inline uint32_t vkd3d_get_priority_adjust(VkDeviceSize size)
+{
+    return min((size / (10 * 1048576)), 0xFFFFUL);
+}
+
+/* map from 32-bit d3d prio to float (0..1) vk prio.*/
+float vkd3d_convert_to_vk_prio(D3D12_RESIDENCY_PRIORITY d3d12prio)
+{
+    float result;
+
+    /* align D3D12_RESIDENCY_PRIORITY_NORMAL (the default d3d12 prio) with
+       0.5 (the default vk prio) so neither kind wins without explicit prio */
+    if (d3d12prio <= D3D12_RESIDENCY_PRIORITY_NORMAL)
+        result = 0.5f * ((double)d3d12prio) / D3D12_RESIDENCY_PRIORITY_NORMAL;
+    else
+        result = 0.5f + 0.5f * ((double)d3d12prio - D3D12_RESIDENCY_PRIORITY_NORMAL) /
+            (UINT32_MAX - D3D12_RESIDENCY_PRIORITY_NORMAL);
```

Comment:
It might be appropriate to clamp the input priority to D3D12_RESIDENCY_PRIORITY_MINIMUM and MAXIMUM to recover some range.

Reply:
vkd3d_get_priority_adjust just steps by 1 ULP every 10 MiB. This will be lost in noise when doing float divisions here. Float based algorithm should take that into account perhaps to add a special priority band just for that perhaps?

I.e.:

MINIMUM to NORMAL -> linear map from 0.1f to 0.5f.
NORMAL to HIGH -> linear map from 0.5f to 0.75f.
HIGH to HIGH + 0x10000 -> linear map from 0.75f to 0.8f
HIGH to MAXIMUM -> linear map from 0.8f to 1.0f.

Reply:
> D3D12_RESIDENCY_PRIORITY_MINIMUM and MAXIMUM
On the one hand it's not an error for an app to submit priorities outside the range, which made me reluctant, but on the other hand that wouldn't be common or smart - so I think that's a good idea.

> HIGH to HIGH + 0x10000 -> linear map from 0.75f to 0.8f

It's a good idea and now that you mention it I'd go further - again although an app can use the full range, the HIGH to HIGH+0x10000 range is particularly special, so I think it should get the lion's share of the 0.5f to 1.0f range to preserve resolution.